Source/WebCore/platform/graphics/coretext/FontCoreText.cpp

```diff
@@ -530,6 +537,62 @@ RefPtr<Font> Font::platformCreateScaledFont(const FontDescription&, float scaleF
     return createDerivativeFont(scaledFont.get(), size, m_platformData.orientation(), fontTraits, m_platformData.syntheticBold(), m_platformData.syntheticOblique(), m_platformData.widthVariant(), m_platformData.textRenderingMode(), m_platformData.protectedCustomPlatformData().get());
 }
 
+bool supportsOpenTypeFeature(CTFontRef font, CFStringRef featureTag)
+{
+    RetainPtr<CFArrayRef> features = adoptCF(CTFontCopyFeatures(font));
+    CFIndex featureCount = CFArrayGetCount(features.get());
+    for (CFIndex featureIndex = 0; featureIndex < featureCount; ++featureIndex) {
+        auto feature = static_cast<CFDictionaryRef>(CFArrayGetValueAtIndex(features.get(), featureIndex));
+        auto featureTypeIdentifierKey = static_cast<CFNumberRef>(CFDictionaryGetValue(feature, kCTFontFeatureTypeIdentifierKey));
+        if (!featureTypeIdentifierKey)
+            continue;
+
+        int featureTypeIdentifier;
+        CFNumberGetValue(featureTypeIdentifierKey, kCFNumberIntType, &featureTypeIdentifier);
+        if (featureTypeIdentifier != kTextSpacingType)
+            continue;
+
+        auto featureSelectors = static_cast<CFArrayRef>(CFDictionaryGetValue(feature, kCTFontFeatureTypeSelectorsKey));
+        if (!featureSelectors)
+            continue;
+        auto selectorsCount = CFArrayGetCount(featureSelectors);
+        for (CFIndex selectorIndex = 0; selectorIndex < selectorsCount; ++selectorIndex) {
+            auto featureSelector = static_cast<CFDictionaryRef>(CFArrayGetValueAtIndex(featureSelectors, selectorIndex));
+            auto openTypeTag = static_cast<CFStringRef>(CFDictionaryGetValue(featureSelector, kCTFontOpenTypeFeatureTag));
+        if (!openTypeTag)
+            continue;
+        if (CFStringCompare(openTypeTag, featureTag, 0))
```

Comment:
This would be better (clearer) written like:

`if (CFStringCompare(openTypeTag, featureTag, 0) ==  kCFCompareEqualTo)`

... unless you want something besides equality.